src/components/BookmarkCard/Details.tsx

```diff
@@ -1,28 +1,73 @@
-import type { PropsWithChildren } from 'react';
+import { PropsWithChildren, useMemo } from 'react';
 import React from 'react';
 import { Link } from '../Link';
 
 import { isEmptyText, stripTags } from '../../lib';
+import classNames from 'classnames';
 
 interface DetailsProps {
     id: string;
     href: string;
     newTab: boolean;
     title?: string;
     description?: string;
+    hasThumbnail: boolean;
+    layout: 'vertical' | 'horizontal';
 }
 
-export function Details({ title, description, href, id, newTab, children }: PropsWithChildren<DetailsProps>) {
+export function Details({
+    title,
+    description,
+    href,
+    id,
+    newTab,
+    layout,
+    hasThumbnail,
+    children,
+}: PropsWithChildren<DetailsProps>) {
+    const { isDescriptionShort, isTitleShort } = useMemo(() => {
+        let titleOneLineLength;
+        let descriptionOneLineLength;
```

Comment:
You can define the default values here and get rid of the `else` block altogether